src/bp-messages/classes/class-bp-messages-notification.php

```diff
@@ -0,0 +1,545 @@
+				'situation_label'     => __( 'A member receives a new private message', 'buddyboss' ),
+				'unsubscribe_text'    => __( 'You will no longer receive emails when someone sends you a message.', 'buddyboss' ),
+			),
+			'bb_messages_new'
+		);
+
+		$this->register_notification(
+			'messages',
+			'bb_messages_new',
+			'bb_messages_new',
+		);
+
+		$this->register_notification_filter(
+			__( 'New messages', 'buddyboss' ),
+			array( 'bb_messages_new' ),
+			30
+		);
+
+		add_filter( 'bp_messages_bb_groups_new_message_notification', array( $this, 'bb_format_messages_notification' ), 10, 7 );
+		add_filter( 'bp_messages_bb_messages_new_notification', array( $this, 'bb_format_messages_notification' ), 10, 7 );
+	}
+
+	/**
+	 * Format the notifications.
+	 *
+	 * @since BuddyBoss [BBVERSION]
+	 *
+	 * @param string $content               Notification content.
+	 * @param int    $item_id               Notification item ID.
+	 * @param int    $secondary_item_id     Notification secondary item ID.
+	 * @param int    $action_item_count     Number of notifications with the same action.
+	 * @param string $format                Format of return. Either 'string' or 'object'.
+	 * @param string $component_action_name Canonical notification action.
+	 * @param string $component_name        Notification component ID.
+	 * @param int    $notification_id       Notification ID.
+	 * @param string $screen                Notification Screen type.
+	 *
+	 * @return array
+	 */
+	public function format_notification( $content, $item_id, $secondary_item_id, $action_item_count, $format, $component_action_name, $component_name, $notification_id, $screen ) {
+		return $content;
+	}
+
+	/**
+	 * Format Messages notifications.
+	 *
+	 * @since BuddyBoss [BBVERSION]
+	 *
+	 * @param string $content               Notification content.
+	 * @param int    $item_id               Notification item ID.
+	 * @param int    $secondary_item_id     Notification secondary item ID.
+	 * @param int    $total_items           Number of notifications with the same action.
+	 * @param string $format                Format of return. Either 'string' or 'object'.
+	 * @param int    $notification_id       Notification ID.
+	 * @param string $screen                Notification Screen type.
+	 *
+	 * @return array
+	 */
+	public function bb_format_messages_notification( $content, $item_id, $secondary_item_id, $total_items, $format, $notification_id, $screen ) {
+
+		$notification = bp_notifications_get_notification( $notification_id );
+		$total_items  = (int) $total_items;
+		$text         = '';
+		$link         = trailingslashit( bp_loggedin_user_domain() . bp_get_messages_slug() . '/inbox' );
+		$amount       = 'single';
+
+		if ( ! empty( $notification ) && 'messages' === $notification->component_name && ( 'bb_groups_new_message' === $notification->component_action || 'bb_messages_new' === $notification->component_action ) ) {
```

Comment:
Do readability format because its long condition
if ( 
			! empty( $notification ) && 
			'messages' === $notification->component_name && 
			( 
				'bb_groups_new_message' === $notification->component_action || 
				'bb_messages_new' === $notification->component_action 
			) 
		) {